components/public-api/gitpod/v1/workspace.proto

```diff
@@ -12,12 +13,33 @@ service WorkspaceService {
   // +return NOT_FOUND User does not have access to a workspace with the given
   // ID +return NOT_FOUND Workspace does not exist
   rpc GetWorkspace(GetWorkspaceRequest) returns (GetWorkspaceResponse) {}
+
+  // ListWorkspaces returns a list of workspaces that match the query.
+  rpc ListWorkspaces(ListWorkspacesRequest) returns (ListWorkspacesResponse) {}
 }
 
 message GetWorkspaceRequest { string id = 1; }
 
 message GetWorkspaceResponse { Workspace item = 1; }
 
+message ListWorkspacesRequest {
+  // pagination contains the pagination options for listing workspaces
+  PaginationRequest pagination = 1;
+
+  // organization_id
+  string organization_id = 2;
+
+  optional bool pinned = 3;
```

Comment:
for the admin case we need other filters as well as sorting. maybe we can generalize somehow?